tests/utils/test_endpoints.py

```diff
@@ -231,3 +231,10 @@ def test_int_arg(value: Optional[Union[int, str]], default: int, expected_result
     if value is not None:
         request.args = {"key": value}
     assert endpoint_utils.int_arg(request, "key", default) == expected_result
+
+
+async def test_endpoint_config_caches_session() -> None:
+    endpoint = endpoint_utils.EndpointConfig("https://example.com/")
+    session = endpoint.session()
+
+    assert session is endpoint.session()
```

Comment:
Can you help me understand this test? It looks like tautology because it does `session = endpoint.session()` and then asserts the same thing in the next line

Reply:
Yes, it's exactly what we're intending, because we've cached the result of `endpoints.session()` with the first method call. And then we need to assert identity, that when we call `endpoints.session()` it doesn't recreate the session, but it returns the same instance.

Reply:
You can try this test by removing the `lru_cache` decorator, and you'll see it fails, with the lru_cache it passes.

Reply:
ah, can you please add a small docstring for this test to explain this. It isn't apparent when reading the test